data-node/src/main/java/org/graylog/datanode/periodicals/MetricsCollector.java

```diff
@@ -59,13 +62,15 @@ public class MetricsCollector extends Periodical {
     private NodeMetricsCollector nodeStatMetricsCollector;
     private ClusterStatMetricsCollector clusterStatMetricsCollector;
     private final ObjectMapper objectMapper;
-    private boolean isLeader;
+
+    private final RestHighLevelClient client;
 
     @Inject
-    public MetricsCollector(OpensearchProcess process, Configuration configuration, ObjectMapper objectMapper) {
+    public MetricsCollector(OpensearchProcess process, Configuration configuration, ObjectMapper objectMapper, RestHighLevelClient client) {
```

Comment:
can't inject the RestHighLevelClient here as it is not managed by guice

Reply:
Good point, fixed by obtaining the client from the already linked process.